packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationBackgroundService.kt

```diff
@@ -0,0 +1,155 @@
+import io.flutter.embedding.engine.dart.DartExecutor
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.view.FlutterCallbackInformation
+import io.flutter.view.FlutterMain
+import java.util.*
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.collections.ArrayDeque
+
+
+class PushNotificationBackgroundService : MethodChannel.MethodCallHandler, JobIntentService() {
+
+    /**
+     * The work queue
+     */
+    private val queue = ArrayDeque<List<Any>>()
+
+    /**
+     * The Background Channel that is used to send messages to the callback dispatcher
+     */
+    private lateinit var backgroundChannel: MethodChannel
+
+    companion object {
+
+        private const val TAG = "PushBackgroundService"
+
+        @JvmStatic
+        private val JOB_ID = UUID.randomUUID().mostSignificantBits.toInt()
+
+        @JvmStatic
+        private var backgroundFlutterEngine: FlutterEngine? = null
+
+        @JvmStatic
+        private val serviceStarted = AtomicBoolean(false)
+
+        @JvmStatic
+        fun enqueueWork(context: Context, work: Intent) {
+            enqueueWork(context, PushNotificationBackgroundService::class.java, JOB_ID, work)
+        }
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        startPushNotificationService(this)
+    }
+
+    private fun startPushNotificationService(context: Context) {
+        synchronized(serviceStarted) {
+            if (backgroundFlutterEngine == null) {
+
+                val callbackHandle = context.getSharedPreferences(
+                    PushNotificationConstants.SHARED_PREFERENCES_KEY, Context.MODE_PRIVATE
+                ).getLong(
+                    PushNotificationConstants.CALLBACK_DISPATCHER_HANDLE_KEY, 0
+                )
+                if (callbackHandle == 0L) {
+                    Log.w(TAG, "Warning: no callback registered")
+                    return
+                }
+                val callbackInfo =
+                    FlutterCallbackInformation.lookupCallbackInformation(callbackHandle)
+                if (callbackInfo == null) {
+                    Log.e(TAG, "Error: failed to find callback")
+                    return
+                }
+                Log.i(TAG, "Starting PushNotificationBackgroundService...")
+
+                // Create a background Flutter Engine
+                backgroundFlutterEngine = FlutterEngine(context)
+
+                val args = DartExecutor.DartCallback(
+                    context.assets, FlutterMain.findAppBundlePath(), callbackInfo
+                )
+
+                // DartCallback must only be executed after the engine has been created
+                backgroundFlutterEngine!!.dartExecutor.executeDartCallback(args)
+            }
+        }
+        backgroundChannel = MethodChannel(
+            backgroundFlutterEngine!!.dartExecutor.binaryMessenger,
+            "plugins.flutter.io/amplify_push_notification_plugin_background"
```

Comment:
Hmm isn't there already an established convention for this within Amplify Flutter?

Reply:
Moved it to the constants file and referenced that.